buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java

```diff
@@ -0,0 +1,449 @@
+import org.gradle.internal.Pair;
+
+/**
+ * Generate HTML report grouped by licenses. Allows to specify a license override file for "Unknown"
+ * license. Allows to specify a multiple license selection file for dependencies which have multiple
+ * licenses available.
+ */
+public class GroupedLicensesHtmlRenderer implements ReportRenderer {
+  private static final String OVERRIDE_PROJECT_URL = "projectUrl";
+  private static final String OVERRIDE_LICENSE = "license";
+  private static final String OVERRIDE_LICENSE_URL = "licenseUrl";
+  private String fileName;
+  private String projectName;
+  private String projectVersion;
+  private File output;
+  private int counter;
+  private int licenseCounter;
+  private final Map<String, Map<String, String>> unknownOverridesLicenses;
+  private final Map<String, String> multiLicenses;
+
+  public GroupedLicensesHtmlRenderer(final Map<String, Object> params) {
+    this(
+        (String) params.getOrDefault("fileName", "index.html"),
+        (String) params.getOrDefault("projectName", null),
+        (File) params.getOrDefault("licenseOverrideFile", null),
+        (File) params.getOrDefault("multiLicensesFile", null));
+  }
+
+  public GroupedLicensesHtmlRenderer(
+      final String fileName,
+      final String projectName,
+      final File licenseOverrideFile,
+      final File multiLicensesFile) {
+    this.fileName = fileName;
+    this.projectName = projectName;
+    if (licenseOverrideFile != null && licenseOverrideFile.exists()) {
+      unknownOverridesLicenses = parseUnknownOverrides(licenseOverrideFile);
+    } else {
+      unknownOverridesLicenses = Collections.emptyMap();
+    }
+
+    if (multiLicensesFile != null && multiLicensesFile.exists()) {
+      multiLicenses = parseMultiLicensesFile(multiLicensesFile);
+    } else {
+      multiLicenses = Collections.emptyMap();
+    }
+  }
+
+  @Input
+  public String getFileNameCache() {
+    return this.fileName;
+  }
+
+  @Override
+  public void render(final ProjectData data) {
+    Project project = data.getProject();
+    projectName = projectName == null ? project.getName() : projectName;
+    projectVersion =
+        !"unspecified".equalsIgnoreCase(project.getVersion().toString())
+            ? project.getVersion().toString()
+            : "";
+    fileName = fileName == null ? "index.html" : fileName;
+    LicenseReportExtension config =
+        project.getExtensions().findByType(LicenseReportExtension.class);
+    if (config == null) {
+      throw new GradleException("LicenseReportExtension is not available");
+    }
+    output = new File(config.outputDir, fileName);
+
+    final Map<ModuleData, Set<Pair<String, String>>> moduleLicenses = buildModuleLicenses(data);
+    final Map<String, List<ModuleData>> licenseGroup = buildLicenseGroup(moduleLicenses);
+
+    writeReport(licenseGroup, moduleLicenses);
+  }
+
+  private Map<ModuleData, Set<Pair<String, String>>> buildModuleLicenses(final ProjectData data) {
```

Comment:
Can you break this up into some smaller methods? There is a lot happening in this method

Reply:
moved into its own repo.